sequence_processing_pipeline/NuQCJob.py

```diff
@@ -396,32 +405,51 @@ def _process_sample_sheet(self):
                 'projects': lst,
                 'sample_ids': sample_ids}
 
+    def _generate_mmi_filter_cmds(self, mmi_db_paths, working_dir):
+        initial_input = join(working_dir, "seqs.r1.fastq")
+        final_output = join(working_dir, "seqs.r1.ALIGN.fastq")
+
+        cmds = []
+
+        tmp_file1 = join(working_dir, "foo")
+        tmp_file2 = join(working_dir, "bar")
+
+        cores_to_allocate = int(self.cores_per_task / 2)
+
+        for count, mmi_db_path in enumerate(mmi_db_paths):
+            if count == 0:
+                # prime initial state with unfiltered file and create first of
+                # two tmp files.
+                input = initial_input
+                output = tmp_file1
+            elif count % 2 == 0:
+                # alternate between two tmp file names so that the input and
+                # output are never the same file.
+                input = tmp_file2
+                output = tmp_file1
+            else:
+                input = tmp_file1
+                output = tmp_file2
+
+            cmds.append(f"minimap2 -2 -ax sr -t {cores_to_allocate} "
+                        f"{mmi_db_path} {input} -a | samtools fastq -@ "
+                        f"{cores_to_allocate} -f 12 -F 256 > {output}")
```

Comment:
Samtools wont benefit from as many cores, better to allocate more to minimap2

Reply:
@antgonza do you have a suggestion on how partition out the cores between the two? Otherwise I can start digging through the documentation.

Reply:
I'll accept it as is for now and we can run some tests first.